gen/property.go

```diff
@@ -0,0 +1,64 @@
+package gen
+
+import (
+	"encoding/json"
+	"fmt"
+	"reflect"
+)
+
+const tagName = "property"
+
+func classify(group interface{}) error {
+	v := reflect.ValueOf(group)
+	if v.Kind() != reflect.Ptr || v.IsNil() {
+		return fmt.Errorf("Can only classify pointers, not %s (%v)", v.Kind(), v)
+	}
+
+	e := v.Elem()
+	t := e.Type()
+
+	var raws []json.RawMessage
+	properties := make(map[string]reflect.Value)
+
+	for i := 0; i < t.NumField(); i++ {
+		tag := t.Field(i).Tag.Get(tagName)
+		field := e.Field(i)
+		if tag == "RAW" {
+			raws = field.Interface().([]json.RawMessage)
+		} else if tag != "" {
+			properties[tag] = field
+		} else {
+			switch field.Kind() {
+			case reflect.Slice:
+				elem := field.Type().Elem()
+				if elem.Kind() == reflect.Struct {
+					for j := 0; j < field.Len(); j++ {
+						classify(field.Index(j).Addr().Interface())
+					}
+				}
+			case reflect.Struct:
+				classify(field.Addr().Interface())
```

Comment:
this line of code not tested in the clock generator... but thought it might necessary for other generators. it's if we have cases nested in other cases